### PR TITLE
Re-enable nocycle linting

### LINKED ISSUE #1622: Re-enable no-cycle linter warnings
**Describe the bug**
In https://github.com/bbc/simorgh/pull/1597 we updated our dependencies but had to add to temporarily disable `import/no-cycle` in `.eslintrc.js`.

This is because with that warning enabled we now get 10 errors of the form:
```
/path/to/simorgh/src/app/containers/Article/index.jsx
  10:1  error  Dependency cycle via ../Text:6=>../Paragraph:2=>../InlineLink:5=>../../routes:10  import/no-cycle
```

We should re-enable the linter rule and fix the errors.

**To Reproduce**

1. Remove `import/no-cycle` from .eslintlrc
2. Run `npm run test:lint`

**Expected behavior**
No linter errors should be shown.

**Screenshots**
NA

**Additional context**

- [x] Initially labelled with ["bug"](https://github.com/bbc/simorgh/labels/bug)

### PR DESCRIPTION
**Pull request by rossgaskell**
Resolves #1622 

**Overall change:** Re-enable no-cycle linting errors

**Code changes:**

- Re-enable linting errors
- Move route regex to separate file

**Investigation**

The errors started appearing due to the bump in eslint import plugin version. There was a bug with the babel-eslint parser not reporting the errors. The errors reported already existed before the dependency update in #1597, but were not reported.

https://github.com/benmosher/eslint-plugin-import/pull/1218
https://github.com/benmosher/eslint-plugin-import/issues/1166

The warnings are a false negative. Whilst there is a circular dependancy due to the internal link requiring the routes regex to match against, this only imports a constant and therefore isn't a true circular dependency.

However to improve readability, the regex paths have been moved to separate file that the routes and internal link component can share without eslint flagging a circular dependency.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

File: src/server/index.jsx [hunk 1]
```diff
@@ -38,6 +39,8 @@ class LoggerStream {
 
 const server = express();
 
+console.log(routes, articleRegexPath);
```
Comment (chris-hinds) [inline src/server/index.jsx]: Console.log left in place

Comment (rossgaskell) [reply src/server/index.jsx]: 🙈

**Comment by jamesdonoh** [comment]
Hi @rossgaskell thanks, this looks good, just for posterity and the next person scratching their head over any similar warnings can you briefly summarise why "Move route regex to separate file" fixed the linter error?

**Review by jamesdonoh — APPROVED** [review]

**Comment by rossgaskell** [comment]
No problem, I'll add my investigation notes from the issue

**Review by chris-hinds — APPROVED** [review]
LGTM, Nice Refactor.

**Comment by PriyaKR** [comment]
Looks good to me.No more linting errors seen when `npm run test:lint` is run.
But there are issues thrown by CC which blocks the merging.

**Pull request merged**